### PR TITLE
chaincode ek message/flow merge

### PR DESCRIPTION
**Pull request by bvavala**
Signed-off-by: Bruno Vavala <bruno.vavala@intel.com>

### CODE REVIEW

File: common/enclave/cc_data.cpp
Comment (g2flyer) [inline common/enclave/cc_data.cpp:40]: Don't have to do it in this PR but we also need the chaincode state encryption key which presumably also would be created here? So a todo might be worthwhile ...

```suggestion
        // TODO: create chaincode state encryption key ...
```

File: common/enclave/cc_data.cpp
Comment (g2flyer) [inline common/enclave/cc_data.cpp:96]: We should explicitly reference that this is a deviation of the specification of FPC lite. 
```suggestion
        // NOTE: for the one-chaincode-one-enclave FPC-Lite version, the chaincode encryption key
        // is serialized directly in the attested data message.
        // This is a (momentary) short-cut over the FPC and FPC Lite specification in `docs/design/fabric-v2+/fpc-registration.puml` and `docs/design/fabric-v2+/fpc-key-dist.puml`

```

File: protos/fpc/fpc.proto
Comment (g2flyer) [inline protos/fpc/fpc.proto:70]: Same as above
```suggestion
    // chaincode encryption key
    // NOTE: This is a (momentary) short-cut over the FPC and FPC Lite specification in `docs/design/fabric-v2+/fpc-registration.puml` and `docs/design/fabric-v2+/fpc-key-dist.puml`
```

**Review by g2flyer — COMMENTED** [review]
From the discussion we had on the short-cut, i would have assumed you also change ERCC to match the change? Should actually work well by changing `QueryChaincodeEncryptionKey` in `ercc/registry/registry.go` by just returning `chaincode_ek` from Credentials.AttestedData?

**Comment by bvavala** [comment]
> From the discussion we had on the short-cut, i would have assumed you also change ERCC to match the change? Should actually work well by changing `QueryChaincodeEncryptionKey` in `ercc/registry/registry.go` by just returning `chaincode_ek` from Credentials.AttestedData?


The implementation of ERCC should actually be completed, and the TODO comments still apply.
I can take care of that in a separate PR -- ideally with additional integration.

**Review by mbrandenburger — APPROVED** [review]
LGTM. Can you add a todo in `ercc/registry/registry.go` inside `RegisterEnclave()` that the chaincode_ek needs to be extracted from attested data and stored under `namespaces/chaincode_ek/<chaincode_id>`

**Comment by g2flyer** [comment]
> LGTM. Can you add a todo in `ercc/registry/registry.go` inside `RegisterEnclave()` that the chaincode_ek needs to be extracted from attested data and stored under `namespaces/chaincode_ek/<chaincode_id>`

Wouldn't you just retrieve it in `QueryChaincodeEncryptionKey` from credential?  I think that would be cleaner as eventually implenting the specs would require only changing the two functions you would expect, i.e., `QueryChaincodeEncryptionKey` and `RegisterCCKeys`?

File: ercc/registry/registry.go
Comment (g2flyer) [inline ercc/registry/registry.go:134]:
```suggestion
        //  NOTE: This is a (momentary) short-cut over the FPC and FPC Lite specification in `docs/design/fabric-v2+/fpc-registration.puml` and `docs/design/fabric-v2+/fpc-key-dist.puml`.  See also `common/enclave/cc_data.cpp` and `protos/fpc/fpc.proto`
        
	// retrieve the enclave id
```

**Review by g2flyer — APPROVED** [review]
Looks good except one minor clickable documentation suggestion.  Looks good to merge once #475 is merged?

**Comment by bvavala** [comment]
Comments addressed, `QueryChaincodeEncryptionKey ` function implemented, travis (and linter!!) doesn't complain.
Merging...

**Pull request merged**
